apps/sms/services/test/unit/message_manager_test.js

```diff
@@ -443,23 +446,29 @@ suite('message_manager.js >', function() {
 
       // doing one more to see if we're not called too many times
       MockNavigatormozMobileMessage.mTriggerMarkReadSuccess();
-      assert.equal(MockNavigatormozMobileMessage.markMessageRead.callCount, 4);
 
-      messagesList.forEach(function(message) {
-        assert.ok(
-          MockNavigatormozMobileMessage.markMessageRead.calledWith(
-            message.id, true
-          )
-        );
-      });
+      Settings.sendReadReport.then(function(result) {
+        messagesList.forEach(function(message) {
+          assert.ok(
+            MockNavigatormozMobileMessage.markMessageRead.calledWith(
+              message.id, true, result
+            )
```

Comment:
nit: you can use `sinon.assert.calledWith(spy, ...)` instead of `assert.ok(spy.calledWith(...))`. It's easier to debug when there is a problem because the error is a lot more verbose.